File: chapter_002/en_ja_dict.py
#!/bin/bash/env python3

import sys

dict_file = "./ejdict-hand-utf8.txt"

if len(sys.argv) != 2:
    print("Usage: {} <word>".format(sys.argv[0]))
    sys.exit(1)
word = sys.argv[1]

with open(dict_file, "r") as f:
    while True:
        line = f.readline()
        if not line:
            break
        if word in line:
            print(line.strip())

File: chapter_002/en_ja_dict.rs
use std::fs::File;
use std::io::{BufRead, BufReader};

fn main() {
    let dict_file = "./ejdict-hand-utf8.txt";

    let args: Vec<String> = std::env::args().collect();

    if args.len() != 2 {
        println!("Usage: {} <word>", args[0]);
        return;
    }

    let word = &args[1].to_lowercase();

    let fp = File::open(dict_file).unwrap();
    let reader = BufReader::new(fp);
    for line in reader.lines() {
        let line = line.unwrap();

        if line.find(word) == None {
            continue;
        }
        println!("{}", line);
    }
}
